src/Formats/SchemaInferenceUtils.cpp

```diff
@@ -324,57 +521,39 @@ namespace
         }
     }
 
-    /// If we have Map and Object(JSON) types, convert all Map types to Object(JSON).
-    /// If we have Map types with different value types, convert all Map types to Object(JSON)
-    void transformMapsAndObjectsToObjects(DataTypes & data_types, TypeIndexesSet & type_indexes)
```

Comment:
I changed logic of experimental JSON type inference. Previously if setting `allow_experimantal_object_type` was enabled, we used `JSON` type for object only if we couldn't infer it as `Map(String, ValueType)`. And I think this logic was not good, no need to use Map type at all, now we always use `JSON` type for objects during inference if `allow_experimantal_object_type` is enabled.